run/helloworld/test/runner.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail;
+
+requireEnv() {
+  test "${!1}" || (echo "Environment Variable '$1' not found" && exit 1)
+}
+requireEnv SERVICE_NAME
+
+# The hello-broken sample needs to be tested with the NAME environment variable
+# both set and unset.
+SERVICE_OVERRIDE="${SERVICE_NAME}-override"
```

Comment:
The hello broken tests deploy two ways to confirm to branching paths the tutorial follows. There are traces of that here and in the system.test.js implementation.

Replacing this with the Pub/Sub implementation may be cleaner, otherwise all the code related to "_OVERRIDE" in the shell script and test should be removed.

Reply:
We did talk about not needing this, though it is nice to confirm that the env var is working. What do you mean by replacing with a Pub/Sub implementation?

Reply:
My thought was maybe in this case it's okay to test the default behavior with just a unit test.

My reference to Pub/Sub is that the bash scripts and system.test.js implementation is 99% the same as what's needed here if we're only doing a single deployment.